src/p4est_io.c

```diff
@@ -2029,7 +2031,8 @@ p4est_file_write_connectivity (p4est_file_context_t * fc,
     p4est_file_write_block (fc, sizeof (size_t),
                             &conn_size_arr,
                             P4EST_STRING " connectivity size", errcode);
-  if (fc == NULL || *errcode != P4EST_FILE_ERR_SUCCESS) {
+  if (*errcode != P4EST_FILE_ERR_SUCCESS) {
+    P4EST_ASSERT (fc == NULL);
     sc_array_destroy (conn_buffer);
     return NULL;
   }
```

Comment:
Would it be even more strict to generally add?

    else { P4EST_ASSERT (fc != NULL); }

Reply:
Yes, I can add this. However, in the relevant code places fc is passed to a p4est_file* function if the error code was P4EST_FILE_ERR_SUCCESS and these functions already assert fc != NULL. Therefore, I think it would be a doubled assertion.

Reply:
Ok, leaving it sounds reasonable.